packages/ffe-searchable-dropdown-react/src/SearchableDropdown.spec.js

```diff
@@ -96,7 +96,7 @@ describe('SearchableDropdown', () => {
 
         userEvent.click(screen.getByText('Bedriften'), { button: 1 });
 
-        expect(onChange).toHaveBeenCalledTimes(1);
+        expect(onChange).toHaveBeenCalledTimes(2);
```

Comment:
Vorfor må denne øke?

Reply:
Siden vi nå i onBlur kjører onChange med selectedItem blir det en ekstra kjøring av denne. 
Hvis vi ikke kjører det ved onBlur, vil vi ikke kjøre det noen gang når selectedItem blir satt ved onBlur, mens når det blir satt ved at vi trykker blir det kjørt i det vi trykker og da onBlurer. Mulig det er en bedre måte å gjøre det på. Kunne kanskje hatt en sjekk i onBlur, men vet ikke helt hva det skulle vært. :thinking:

Reply:
Jag tror vi burde implementera noen form a throttling  eller bruke en ferdig https://underscorejs.org/docs/modules/throttle.html

Reply:
Hvorfor trenger vi det her? Holder det ikke med de endringene vi har? 
Vi har sagt til bankene at vi skal ut med kontovelgeren denne uka, og det her er det eneste vi mangler, så hadde vært fint om vi kunne hatt denne lille løsninga midlertidig hvert fall. Så kan vi senere implementere throttle? :D 
Litt fint om vi får det her ut så fort som mulig.

Reply:
Jeg synes det er skummelt å skulle gå inn og rote med ting nå som det nesten er klart. Kunne vi midlertidig gått ut uten det hvis det kommer til å bli styr?

Reply:
Nu har jag ikke provat selv men tenkte i så fall noe lignande? Men vet ikke hvordan det slår ut på testerna :)

```
import  throttle from 'lodash.throttle'


const SearchableDropdown = ({ onChange }) => {
 ......
     const throttledOnChange= useCallback(throttle(onChange, 300) , [onChange]);
   ......
    if (hasEmptiedInputFieldWithoutClearButton) {
            dispatch({ type: stateChangeTypes.ClearedInputField });
            throttledOnChange(null);
    }
}


```

Reply:
Aha, da prøver jeg på noe sånt.
Men hva slags forskjell vil de utgjøre egentlig?

Reply:
Det hjalp ikke på testene. :/ Tenker du det fortsatt er lurt å gjøre det sånn uansett eller? :)

Reply:
Hvis du tester litt manuellt og det ser greit ut. Ha en console.log i onChange så ær det nok bedre tenker jag iaf

Reply:
før du bytte ut alle onChange til throttledOnChange og ikke kun den jag vise i exemplet?

Reply:
Forskjellen blir ju att det maks blir en onChange varej 300 ms istellet før n. Jag vet ikke om det har noen praktiskt betydning før dom som bruker det men jag synes egentlig onChange skall bli kallad en gang per endrad verdi. 

En annen mulig løsning kanske vill vart og huske verdien man siste brukte som input til onChange og hvis den er detsamme ikke kalla på onChange.

Reply:
Åja, jeg satte ikke på alle onChangene. Gjorde det kun i den funksjonen. Men da prøver jeg på det :smile:

Reply:
Jag tror egentlig jag liker siste forslaget bedre

```
const lastValue = useRef(null/initialValue)

const handleChange = (value) => {
      if(!equal(value, lastValue.current ) ) {
          onChange(value);
          lastValue.current = value
      }
}


```

Jag vet ikke vad dere tenker men synes det blir bedre selv. Jag synes det er ganska viktigt att den kun blir kallt en gang per endring hvis noen feks har tracking på onChange

Reply:
Enig i at det kanskje er bedre å heller huske forrige verdi enn å bruke throttle/debounce, siden man i sistnevnte tilfelle kunne gått glipp av onChange-events der verdiene faktisk har blitt endret.

Er det forresten noen spesiell grunn til å bruke `useRef` i steden for `useState` i eksempelet ditt @pethel? Er det for å unngå re-rendring ved endring av state?

Reply:
Jeg har funnet en måte vi slipper å bruke throttle på :blush: Pusher det når det er klart, så kan dere ta en titt og se hva dere synes.

Reply:
Enda bedre

Reply:
@fossecode  ja ser ikke noen grund til og bruke en state her. 

https://reactjs.org/docs/hooks-reference.html#useref
![image](https://user-images.githubusercontent.com/2248579/123067003-e0bd4e00-d410-11eb-84e6-c988c7cccd4b.png)

Reply:
Prata med Kristofer og fant en løsning som funka med throttle. Så det er bare å ta en titt og se hva dere synes om hvordan det ser ut nå :blush:

Reply:
Prata med Kristofer og fant en løsning med throttle. Men ser nå at testene i AccountSelector feiler fordi onChange nå blir satt litt senere enn forventa. Det tyder kanskje på at mange andre vil slite med å lage tester når de tar i bruk SearchableDropdown og AccountSelector. :/

Reply:
Forhåpentligvis siste nå! :sweat_smile: Fant en løsning uten throttle :smile: